test/control.js

```diff
@@ -79,3 +79,58 @@ metatests.test('sequential with error', test => {
     test.end();
   });
 });
+
+metatests.test('runIf run asyncFn', test => {
+  const asyncFn = test.mustCall((arg1, arg2, cb) => {
+    process.nextTick(() => {
+      cb(null, { arg1, arg2 });
+    });
+  });
+
+  metasync.runIf(true, asyncFn, 'val1', 'val2', (err, result) => {
+    test.error(err);
+    test.strictSame(result, { arg1: 'val1', arg2: 'val2' });
+    test.end();
+  });
+});
+
+metatests.test('runIf do not run asyncFn', test => {
+  const asyncFn = test.mustNotCall((arg1, arg2, cb) => {
+    process.nextTick(() => {
+      cb(null, { arg1, arg2 });
+    });
+  });
+
+  metasync.runIf(false, asyncFn, 'val1', 'val2', (err, result) => {
+    test.error(err);
+    test.strictSame(result, undefined);
+    test.end();
+  });
+});
+
+metatests.test('runIf default value', test => {
+  const asyncFn = (val, cb) => {
```

Comment:
I think this can just be `const asyncFn = test.mustNotCall()`.

Reply:
ping @o-rumiantsev

Reply:
Sorry. I think it is better to leave it as it is now.

Reply:
@o-rumiantsev why?

Reply:
@nechaido think, this can be less understandable

Reply:
@o-rumiantsev can you at least wrap this function in `test.mustNotCall()` like you did in the test above this one (`'runIf do not run asyncFn'`)?

Reply:
@belochub ofc